pandas/core/groupby/groupby.py

```diff
@@ -2675,6 +2674,51 @@ def rank(
         Returns
         -------
         DataFrame with ranking of values within each group
+
+        See Also
+        --------
+        Series.groupby : Apply a function groupby to a Series.
+        DataFrame.groupby : Apply a function groupby
+            to each row or column of a DataFrame.
+        Series.rank : Apply a function rank to a Series.
+        DataFrame.rank : Apply a function rank
+            to each row or column of a DataFrame.
+
+        Examples
+        --------
+        >>> df = pd.DataFrame({'group': ['a', 'a', 'a', 'b',
+        ...                              'a', 'b', 'b', 'b', 'b', 'a'],
+        ...                    'value': [.2, .4, .2, 0.01,
+        ...                              .3, .11, .21, .4, .01, 0.2]})
+        >>> df
+          group  value
+        0     a   0.20
+        1     a   0.40
+        2     a   0.20
+        3     b   0.01
+        4     a   0.30
+        5     b   0.11
+        6     b   0.21
+        7     b   0.40
+        8     b   0.01
+        9     a   0.20
+        >>> df['average_rank'] = df.groupby('group')['value'].rank('average')
+        >>> df['min_rank'] = df.groupby('group')['value'].rank('min')
+        >>> df['max_rank'] = df.groupby('group')['value'].rank('max')
+        >>> df['dense_rank'] = df.groupby('group')['value'].rank('dense')
+        >>> df['first_rank'] = df.groupby('group')['value'].rank('first')
```

Comment:
This might be clearer as is, but could be written more concisely along the lines of
```
for method in ["average", ..., "first"]:
   df[f"{method}_rank"] = df.groupby("group")["value"].rank(method)
```